test/threads.jl

```diff
@@ -141,3 +141,41 @@ end
 @test_throws TypeError Atomic{Bool}
 @test_throws TypeError Atomic{BigInt}
 @test_throws TypeError Atomic{Float64}
+
+# Test atomic fences
+type CommBuf
+    barrier::Atomic{Int}
+    data::Vector{Float64}
+    correct::Bool
+    CommBuf() = new(Atomic{Int}(0), Vector{Float64}(100), false)
+end
+function test_fence_write(commbuf::CommBuf)
+    sleep(0.1)
```

Comment:
what's complicated about it? It's literally two lines of code (one for unix one for windows) each with a single ccall and no loops. https://github.com/JuliaLang/julia/blob/7b1cba7fb85e590666cce17dcaa85971455b5b9b/test/ccall.jl#L507